consensus/core/src/tests/base_committer_tests.rs

```diff
@@ -0,0 +1,732 @@
+    let leader_status = committer.try_direct_decide(leader_wave_2);
+    info!("Leader commit status: {leader_status}");
+
+    if let LeaderStatus::Undecided(undecided_slot) = leader_status {
+        assert_eq!(undecided_slot, leader_wave_2)
+    } else {
+        panic!("Expected an undecided leader")
+    };
+
+    // 3. Ensure we skip leader of wave 2 indirectly.
+    info!("Try indirect commit for leader {leader_wave_2}",);
+    let leader_status = committer.try_indirect_decide(leader_wave_2, decided_leaders.iter());
+    info!("Leader commit status: {leader_status}");
+
+    if let LeaderStatus::Skip(skipped_slot) = leader_status {
+        assert_eq!(skipped_slot, leader_wave_2)
+    } else {
+        panic!("Expected a skipped leader")
+    };
+
+    // Ensure we directly commit the leader of wave 1.
+    let leader_round_wave_1 = wave_length;
+    let leader_wave_1 = committer
+        .elect_leader(leader_round_wave_1)
+        .expect("should have elected leader");
+    info!("Try direct commit for leader {leader_wave_1}");
+    let leader_status = committer.try_direct_decide(leader_wave_1);
+    info!("Leader commit status: {leader_status}");
+
+    if let LeaderStatus::Commit(ref committed_block) = leader_status {
+        assert_eq!(committed_block.author(), leader_wave_1.authority);
+    } else {
+        panic!("Expected a committed leader")
+    };
+}
+
+/// If there is no leader with enough support nor blame, we commit nothing.
+#[test]
+fn undecided() {
+    telemetry_subscribers::init_for_testing();
+    // Commitee of 4 with even stake
+    let context = Arc::new(Context::new_for_test(Some(4)));
+    let dag_state = Arc::new(RwLock::new(DagState::new(
+        context.clone(),
+        Arc::new(MemStore::new()),
+    )));
+    let committer = BaseCommitterBuilder::new(context.clone(), dag_state.clone()).build();
+
+    let wave_length = DEFAULT_WAVE_LENGTH;
+
+    // Add enough blocks to reach the leader round of wave 1.
+    let leader_round_wave_1 = wave_length;
+    let references_leader_round_wave_1 = build_dag(
+        context.clone(),
+        dag_state.clone(),
+        None,
+        leader_round_wave_1,
+    );
+
+    // Filter out that leader.
+    let leader_wave_1 = committer
+        .elect_leader(leader_round_wave_1)
+        .expect("should have elected leader");
+    let references_without_leader_wave_1: Vec<_> = references_leader_round_wave_1
+        .iter()
+        .cloned()
+        .filter(|x| x.author != leader_wave_1.authority)
+        .collect();
+
+    // Create a dag layer where only one authority votes for the leader of wave 1.
+    let mut authorities = context.committee.authorities();
+    let connections_leader_wave_1 = vec![(
+        authorities.next().unwrap().0,
+        references_leader_round_wave_1,
+    )];
+    let connections_without_leader_wave_1: Vec<_> = authorities
+        .take((context.committee.quorum_threshold() - 1) as usize)
```

Comment:
Should this be `authorities.skip(1)`? Otherwise I wonder if the same authority has multiple blocks.

Reply:
I believe the first authority is filtered out from the `authorities` iterator on line 525
```
authorities.next().unwrap().0
``` 

so it has already been "skipped". So here I would expect the `.take((context.committee.quorum_threshold() - 1) as usize)` to give us the next 2 authorities (the ones in position 1 & 2). Why not just take all the rest of them?